stream/hls.go

```diff
@@ -25,18 +31,19 @@ type HLSMuxer interface {
 
 //TODO: Write tests, set buffer size, kick out segments / playlists if too full
 type HLSBuffer struct {
-	HoldTime   time.Duration
 	plCacheNew bool
-	segCache   *Queue
+	segCache   []string
 	// pq       *Queue
-	plCache m3u8.MediaPlaylist
-	sq      *ConcurrentMap
-	lock    sync.Locker
+	plCache  m3u8.MediaPlaylist
+	sq       *ConcurrentMap
+	lock     sync.Locker
+	Capacity uint
 }
 
-func NewHLSBuffer() *HLSBuffer {
```

Comment:
Golang doesn't have method overload, which makes things a little annoying to maintain (cannot just add a new param in the constructor without changing existing code).  Not sure what's the best practice here.  I can add a "NewHLSBufferWithCap", but maybe it's good to enforce the cap for this case (following the same pattern as grafov/m3u8).